server/service/apple_mdm.go

```diff
@@ -982,62 +940,10 @@ func (svc *Service) EnqueueMDMAppleCommand(
 	if err != nil {
 		err = fleet.NewInvalidArgumentError("command", "unable to decode base64 command").WithStatus(http.StatusBadRequest)
 
-		return 0, nil, ctxerr.Wrap(ctx, err, "decode base64 command")
-	}
-	cmd, err := mdm.DecodeCommand(rawXMLCmd)
-	if err != nil {
-		err = fleet.NewInvalidArgumentError("command", "unable to decode plist command").WithStatus(http.StatusUnsupportedMediaType)
-		return 0, nil, ctxerr.Wrap(ctx, err, "decode plist command")
+		return nil, ctxerr.Wrap(ctx, err, "decode base64 command")
 	}
 
-	if premiumCommands[strings.TrimSpace(cmd.Command.RequestType)] {
-		lic, err := svc.License(ctx)
-		if err != nil {
-			return 0, nil, ctxerr.Wrap(ctx, err, "get license")
-		}
-		if !lic.IsPremium() {
-			return 0, nil, fleet.ErrMissingLicense
-		}
-	}
-
-	if err := svc.mdmAppleCommander.EnqueueCommand(ctx, deviceIDs, string(rawXMLCmd)); err != nil {
-		// if at least one UUID enqueued properly, return success, otherwise return
-		// error
-		var apnsErr *apple_mdm.APNSDeliveryError
-		var mysqlErr *mysql.MySQLError
-		if errors.As(err, &apnsErr) {
-			if len(apnsErr.FailedUUIDs) < len(deviceIDs) {
-				// some hosts properly received the command, so return success, with the list
-				// of failed uuids.
-				return http.StatusOK, &fleet.CommandEnqueueResult{
-					CommandUUID: cmd.CommandUUID,
-					RequestType: cmd.Command.RequestType,
-					FailedUUIDs: apnsErr.FailedUUIDs,
-				}, nil
-			}
-			// push failed for all hosts
-			err := fleet.NewBadGatewayError("Apple push notificiation service", err)
-			return http.StatusBadGateway, nil, ctxerr.Wrap(ctx, err, "enqueue command")
-
-		} else if errors.As(err, &mysqlErr) {
-			// enqueue may fail with a foreign key constraint error 1452 when one of
-			// the hosts provided is not enrolled in nano_enrollments. Detect when
-			// that's the case and add information to the error.
-			if mysqlErr.Number == mysqlerr.ER_NO_REFERENCED_ROW_2 {
-				err := fleet.NewInvalidArgumentError(
-					"device_ids",
-					fmt.Sprintf("at least one of the hosts is not enrolled in MDM or is not a macOS device: %v", err),
-				).WithStatus(http.StatusBadRequest)
-				return http.StatusBadRequest, nil, ctxerr.Wrap(ctx, err, "enqueue command")
-			}
-		}
-
-		return http.StatusInternalServerError, nil, ctxerr.Wrap(ctx, err, "enqueue command")
-	}
-	return http.StatusOK, &fleet.CommandEnqueueResult{
-		CommandUUID: cmd.CommandUUID,
-		RequestType: cmd.Command.RequestType,
-	}, nil
+	return svc.enqueueAppleMDMCommand(ctx, rawXMLCmd, deviceIDs)
```

Comment:
There's an exported method with the same name [above](https://github.com/fleetdm/fleet/pull/14837/files#diff-1cc547279489e5119326f2ac15610c2dd7519ef7dada5952552765cefabd41aaR922). Should that exported method be deleted in favor of the unexported one added to the shared mdm file [below](https://github.com/fleetdm/fleet/pull/14837/files#diff-1cc547279489e5119326f2ac15610c2dd7519ef7dada5952552765cefabd41aaR922)?

Reply:
the exported method is used by the deprecated endpoint, I think we still need it, right? Probably I'm not understanding what you meant

Reply:
I think that's my change, what happens here is that `EnqueueMDMAppleCommand` backs the deprecated endpoint while `enqueueAppleMDMCommand` is the common implementation of queueing an Apple command by both the deprecated endpoint and the new platform-agnostic one (just now realizing that one is MDMApple and the other is AppleMDM but oh well...).